algorithm/index.mjs

```diff
@@ -1,17 +1,487 @@
+      totalInitialCosts += extraCost.value;
+    } else if (extraCost.type === "Inicial" && extraCost.valueType === "Porcentual") {
+      totalInitialCosts += (extraCost.value / 100) * inputs.sellingPrice;
+    } else if (extraCost.type === "Periódico" && extraCost.valueType === "Monetario") {
+      periodicCosts -= extraCost.value;
+    } else {
+      insuranceAmount = -(((extraCost.value / 100) * newSellingPrice) / cuotasAnuales);
+    }
+  }
+
+  const leasingAmount = totalInitialCosts + sellingValue;
+
+  let periodicalInterestRate = 0;
+
+  if (inputs.interestRateType === "Nominal") {
+    const m = inputs.interestRateFrequency / inputs.capitalizacion;
+    const n = inputs.paymentFrequency / inputs.capitalizacion;
+    periodicalInterestRate = Math.pow(1 + inputs.interestRate / 100 / m, n) - 1;
+  } else {
+    periodicalInterestRate =
+      Math.pow(1 + inputs.interestRate / 100, inputs.paymentFrequency / inputs.interestRateFrequency) - 1;
+  }
+
+  console.log(`\nPrecio de venta del activo: S/ ${inputs.sellingPrice}`);
+  console.log(`Cuota inicial: S/ ${initialFee}\n`);
+  console.log(`Nuevo precio de venta S/ ${newSellingPrice}`);
+  console.log(`Monto de IGV: S/ ${montoIGV}`);
+  console.log(`Valor de venta del activo: S/ ${sellingValue}`);
+  console.log(`Total por costos iniciales: S/ ${totalInitialCosts}`);
+  console.log(`Monto del leasing: S/ ${leasingAmount}`);
+
+  console.log(`\nN° cuotas al año: ${cuotasAnuales}`);
+  console.log(`N° periodos de pago: ${periodos}`);
+  console.log(`Frecuencia de pago: ${inputs.paymentFrequency} días\n`);
+
+  console.log(`Tipo de tasa de interés: ${inputs.interestRateType}`);
+  console.log(`Tasa efectiva del periodo: ${periodicalInterestRate * 100}%\n`);
+
+  console.log(`Tasa descuento Ks equivalente: ${descuentoKs * 100}%`);
+  console.log(`Tasa descuento WACC equivalente: ${descuentoWACC * 100}%`);
+
+  // Cronograma de pagos
+
+  let tipoPlazo = "";
+  let saldoInicial = leasingAmount;
+  let cuota = 0;
+  let intereses = 0;
+  let amortizacion = 0;
+  let saldoFinal = 0;
+  const depreciacion = -(sellingValue / periodos);
+  let ahorroTributario = 0;
+  let montoIGVPeriodico = 0;
+  let flujoBruto = 0;
+  let flujoIGV = 0;
+  let flujoNeto = 0;
+
+  let totalIntereses = 0;
+  let totalAmortizacion = 0;
+  let totalCostosPeriodicos = 0;
+  let totalSeguro = 0;
+
+  let vnaFlujoBruto = 0;
+  let vnaFlujoNeto = 0;
+
+  const flujosBrutos = [leasingAmount];
+  const flujosNetos = [leasingAmount];
+
+  for (let periodo = 1; periodo <= periodos; periodo++) {
+    console.log(`= = = = = Periodo ${periodo} = = = = =`);
+
+    intereses = -(periodicalInterestRate * saldoInicial);
+
+    if (periodo < periodos) {
+      const input = await inquirer.prompt({
+        name: "plazoDeGracia",
+        type: "list",
+        message: `Escoja el tipo de plazo de gracia: `,
+        choices: ["Total", "Parcial", "Sin plazo de gracia"],
+      });
```

Comment:
Is this better than trying to implement the [table prompt](https://github.com/eduardoboucas/inquirer-table-prompt) we talked about in Telegram?